hooks/context_change.py

```diff
@@ -17,8 +17,14 @@
 
 class ContextChangeHook(get_hook_baseclass()):
     """
-    Hook that gets executed every time there is a context change, once before
-    the change and once after.
+    Hook that gets executed every single time there is a context change in Toolkit,
+    e.g. when an engine starts or restarts and when sgtk.platform.change_context
+    is invoked.
```

Comment:
As a lazy developer, I would love if there were bullet points. Something along these lines:


- If an engine *starts up*, the ``current_context`` passed to the hook methods will be None and the ``next_context`` parameter will be set to the context that the engine is starting up.
- If an engine is *reloading itself*, for example someone developer has hit the refresh button, the ``current_context`` and ``next_context`` parameters will contain the same object.
- If a *context switch is requested*, for example when a user switches from project to shot mode in Nuke Studio, ``current_context`` and ``next_context`` will contain two different contexts.